src/core/service.c

```diff
@@ -1256,10 +1256,17 @@ static int service_spawn(
                 socklen_t salen = sizeof(sa);
 
                 r = getpeername(s->socket_fd, &sa.sa, &salen);
-                if (r < 0)
-                        return -errno;
+                if (r < 0) {
+                        r = -errno;
+
+                        /* ENOTCONN is legitimate if the endpoint disappeared on shutdown.
+                         * This connection is over, but the socket unit lives on. */
+                        if (r != -ENOTCONN ||
+                            (c != s->exec_command[SERVICE_EXEC_STOP] && c != s->exec_command[SERVICE_EXEC_STOP_POST]))
```

Comment:
oh, this doesn't work if there are two (or more) `ExecStop`-commands. Should be
```sh
--- a/src/core/service.c
+++ b/src/core/service.c
@@ -1261,8 +1261,7 @@ static int service_spawn(

                         /* ENOTCONN is legitimate if the endpoint disappeared on shutdown.
                          * This connection is over, but the socket unit lives on. */
-                        if (r != -ENOTCONN ||
-                            (c != s->exec_command[SERVICE_EXEC_STOP] && c != s->exec_command[SERVICE_EXEC_STOP_POST]))
+                        if (r != -ENOTCONN || !IN_SET(s->control_command_id, SERVICE_EXEC_STOP, SERVICE_EXEC_STOP_POST))
                                 return r;
                 }
```
@bkylerussell , would you like to send a patch?